sweepai/api.py

```diff
@@ -234,6 +193,82 @@ def call_on_merge(*args, **kwargs):
     thread.start()
 
 
+function_mapping = {
+    "on_merge": on_merge,
+    "on_merge_conflict": on_merge_conflict,
+    "stack_pr": stack_pr,
+    "make_pr": make_pr,
+    "handle_button_click": handle_button_click,
+}
+
+
+@dataclass
+class Task:
+    handler_name: str
+    args: tuple = ()
+    kwargs: dict = field(default_factory=dict)
+
+
+user_queues: dict[str, "Queue[Task]"] = SafeDictionary()
+active_workers: dict[str, 0] = SafeDictionary()
+
+
+def display_queue():
+    result = "Queue:\n"
+    for key, q in list(user_queues.items()):
+        items = ", ".join([task.handler_name for task in q.queue])
+        result += f"{key}: {items}\n"
+    result += str(active_workers._dict)
+    return result
```

Comment:
**Optimize the display_queue function for better performance using list comprehension and string joining.** 
 
 Implement a more efficient queue display function using list comprehension and string joining for better performance. 
```suggestion
def display_queue():
    queue_items = [
        f"{key}: {', '.join(task.handler_name for task in q.queue)}"
        for key, q in user_queues.items()
    ]
    return "Queue:
" + "
".join(queue_items) + f"
{active_workers._dict}"
```